api/gateway/v1alpha1/gateway_types.go

```diff
@@ -107,13 +107,27 @@ func (gw *Gateway) Validate(_ context.Context, _ kclient.Reader) error {
 	if !vtepIP.Addr().Is4() {
 		return fmt.Errorf("VTEPIP %s must be an IPv4 address", gw.Spec.VTEPIP) //nolint:goerr113
 	}
+	if vtepIP.Addr().IsMulticast() || vtepIP.Addr().IsLoopback() || vtepIP.Addr().IsUnspecified() {
+		return fmt.Errorf("VTEPIP %s must be a unicast IPv4 address", gw.Spec.VTEPIP) //nolint:goerr113
+	}
+	localhostNet := netip.MustParsePrefix("127.0.0.0/8")
```

Comment:
I really don't like using functions that are designed to panic instead of returning errors, as if we refactor that code and it becomes an invalid value at some point, we may end up with a panic in runtime...